src/Microsoft.SqlTools.ServiceLayer/Metadata/MetadataScriptCacher.cs

```diff
@@ -0,0 +1,108 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Logger = Microsoft.SqlTools.Utility.Logger;
+
+namespace Microsoft.SqlTools.ServiceLayer.Metadata
+{
+    public static class MetadataScriptCacher
```

Comment:
It's not clear to me how this is supposed to be used - there doesn't seem to be any sort of cache expiration or validation that the cache isn't valid so right now it seems completely up to the client side to determine when to refresh the cache which doesn't seem like an easy thing to do